apps/trading/e2e/tests/wallet/test_wallet.py

```diff
@@ -0,0 +1,116 @@
+import pytest
+import re
+import json
+from playwright.sync_api import Page, expect, Route
+from vega_sim.service import VegaService
+from conftest import init_vega
+from fixtures.market import setup_continuous_market
+
+order_size = "order-size"
+order_price = "order-price"
+place_order = "place-order"
+order_side_sell = "order-side-SIDE_SELL"
+market_order = "order-type-Market"
+tif = "order-tif"
+expire = "expire"
+
+
+@pytest.fixture(scope="module")
+def vega(request):
+    with init_vega(request) as vega:
+        yield vega
+
+
+@pytest.fixture(scope="module")
+def continuous_market(vega):
+    return setup_continuous_market(vega)
+
+def handle_route_connection_lost(route: Route, request):
+        if request.method == "POST" and re.match(r"http://localhost:\d+/api/v2/requests", request.url):
+            route.fulfill(
+                status=200,
+                headers={"Content-Type": "application/json"},
+                body='{"jsonrpc": "2.0", "id": "1"}'
+            )
+        else:
+            route.continue_()
+
+def handle_route_connection_rejected(route: Route, request):
+        if request.method == "POST" and re.match(r"http://localhost:\d+/api/v2/requests", request.url):
+            custom_response = {
+                "jsonrpc": "2.0",
+                "error": {
+                    "code": 3001,
+                    "data": "the user rejected the wallet connection",
+                    "message": "User error"
+                },
+                "id": "0"
+            }
+            route.fulfill(
+                status=400,
+                headers={"Content-Type": "application/json"},
+                body=json.dumps(custom_response)
+            )
+        else:
+            route.continue_()
+
+def assert_connection_approve(route: Route, request, page:Page):
+        if request.method == "POST" and re.match(r"http://localhost:\d+/api/v2/requests", request.url):
+            expect(page.get_by_test_id("toast-content")).to_have_text("Please go to your Vega wallet application and approve or reject the transaction.")
+        else:
+            route.continue_()
+
+@pytest.mark.usefixtures("page", "auth", "risk_accepted")
+def test_wallet_connection_error(continuous_market, page: Page):
+    page.goto(f"/#/markets/{continuous_market}")
```

Comment:
Personal preference and I know it is a one liner so will leave with you, I literally see this line everywhere and I would just have this url declared in one place. Or enclose it in a method where you can pass in any market. If this url changes for any reason (although I am assuming unlikely) I would not want to change in 56 tests.